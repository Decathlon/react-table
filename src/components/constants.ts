export const DEFAULT_ROW_HEIGHT = 56;

export const DEFAULT_COLUMN_WIDTH = "auto";

export const DEFAULT_COLSPAN = 1;

export const BORDER_SIZE = 1;

export const MIN_COLUMN_WIDTH = 100;

export const SUBCELL_PADDING = 30;

// must have the same value as max-row-level in style/table/_row.scss
export const MAX_ROW_LEVEL = 2;

export const ROW_SPAN_WIDTH = 110;

export enum MouseClickButtons {
  right = "right",
  left = "left"
}

export const RowHeight = {
  small: 60,
  medium: 80,
  large: 100
};

export const ColumnWidth = {
  small: 90,
  medium: 160,
  large: 230
};
