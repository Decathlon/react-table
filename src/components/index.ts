export { default as ResponsiveContainer } from "./responsive-container";
export * from "./responsive-container";

export { default as Scroller } from "./scroller";
export * from "./scroller";

export { default as Virtualizer } from "./virtualizer";
export * from "./virtualizer";

export * from "./styled-table";
export * from "./table";
export * from "./table-interactions-manager";
export * from "./table-selection";
export * from "./utils";
export * from "./constants";
export * from "./typing";
