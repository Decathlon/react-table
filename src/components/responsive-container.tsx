import * as React from "react";
import classNames from "classnames";

interface IElementSize {
  width: number;
  height: number;
}

export interface IResponsiveContainerOptionalProps {
  className?: string;
}

export interface IResponsiveContainerProps extends IResponsiveContainerOptionalProps {
  children: (size: IElementSize) => JSX.Element;
}

const ResponsiveContainer = ({ className, children }: IResponsiveContainerProps) => {
  const wrapper = React.useRef<HTMLDivElement>(null);
  const [{ height, width }, setSize] = React.useState<IElementSize>({
    height: 0,
    width: 0
  });

  const updateSize = () => {
    if (wrapper && wrapper.current) {
      const { clientWidth, clientHeight } = wrapper.current;
      if (clientWidth !== width || clientHeight !== height) {
        setSize({
          width: clientWidth,
          height: clientHeight
        });
      }
    }
  };

  React.useEffect(() => {
    updateSize();
    window.addEventListener("resize", updateSize);
    return () => {
      window.removeEventListener("resize", updateSize);
    };
  }, [width, height]);

  return (
    <div className={classNames("responsive-container", className)} ref={wrapper}>
      {width && height ? children({ width, height }) : null}
    </div>
  );
};

export default ResponsiveContainer;
