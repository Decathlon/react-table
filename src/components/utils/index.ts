export * from "./table";
export * from "./table-selection";
export * from "./common";

export { default as shallowEqual } from "./shallowEqual";
