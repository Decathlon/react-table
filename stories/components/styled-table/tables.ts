import { IRow } from "../../../src/components/table/row";
import HeaderCell from "../../../src/components/styled-table/header-cell";

const indicatorNames = [
  "CA_TTC_OMNICANAL",
  "PROG_CA_TTC",
  "MARGIN_RATE",
  "HOURS",
  "FEES_PROGRESSION",
  "TURNOVER",
  "RATE_SHRINKAGE",
  "PERSONAL_FEES",
  "SQUARE_METERS",
  "SURFACE_COST",
  "TOTAL_FEES",
  "CONTRIBUTION",
  "OVERHEADS",
  "ADVISED_TURNOVER",
  "CA_TTC_OMNICANALF",
  "PROG_CA_TTCF",
  "MARGIN_RATEF",
  "HOURSF",
  "FEES_PROGRESSIONF",
  "TURNOVERF",
  "RATE_SHRINKAGEF",
  "PERSONAL_FEESF",
  "SQUARE_METERSF",
  "SURFACE_COSTF",
  "TOTAL_FEESF",
  "CONTRIBUTIONF",
  "OVERHEADSF",
  "ADVISED_TURNOVERF",
];

const text = "Lorem Ipsum is that it has a more-or-less normal";

export function getTable(cells = {}) {
  const weeks = Array.from(Array(52), (_, rowIndex) => ({
    id: rowIndex < 9 ? `0${rowIndex + 1}` : `${rowIndex + 1}`,
    value: `W${rowIndex + 1}${rowIndex === 3 ? text : ""}`,
  }));
  const headerRow: IRow = {
    id: "headers",
    isHeader: true,
    size: 126,
    cells: [
      {
        id: "indicators",
        value: `Indicateurs ${text}`,
      },
    ],
  };
  weeks.forEach((week) =>
    headerRow.cells.push({
      id: week.id,
      value: week.value,
      cellContent: HeaderCell,
      cellContentProps: {
        title: "2019",
        isCurrent: week.value === "W12",
        subtitle: "1 éphéméride",
        value: week.value,
      },
    })
  );
  const rows: IRow[] = [headerRow];

  indicatorNames.forEach((indicatorName) => {
    const row: IRow = {
      id: indicatorName,
      cells: [
        {
          id: indicatorName,
          value: indicatorName,
        },
      ],
    };
    weeks.forEach((week) => {
      row.cells.push({
        id: `${indicatorName}-${week.value}`,
        value: "0.00",
      });
    });
    rows.push(row);
  });

  Object.keys(cells).forEach((row) => {
    Object.keys(cells[row]).forEach((cell) => {
      rows[row].cells[cell] = cells[row][cell];
    });
  });

  return { id: "styled-tabel", rows };
}
