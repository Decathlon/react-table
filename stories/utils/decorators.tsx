import { ThemeProvider, StyledEngineProvider, createTheme, adaptV4Theme } from "@mui/material/styles";

export const blueDkt = "#0082c3";

const muiDktTheme = createTheme(
  adaptV4Theme({
    typography: {
      fontFamily: "Roboto Condensed",
    },
    palette: {
      primary: {
        light: blueDkt,
        dark: blueDkt,
        main: blueDkt,
      },
      secondary: {
        light: blueDkt,
        dark: blueDkt,
        main: blueDkt,
      },
    },
  })
);

/**
 * The MUI theme provider wrapper
 *
 * @param {() => JSX.Element} attributes a function to render the child component.
 *
 * @return {JSX.Element} The child wrapped by the MUI theme provider.
 */
export function withThemeProvider(renderChild: () => JSX.Element): JSX.Element {
  return (
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={muiDktTheme}>{renderChild()}</ThemeProvider>
    </StyledEngineProvider>
  );
}
