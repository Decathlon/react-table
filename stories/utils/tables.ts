import { IRow } from "../../src/components/table/row";

export const simpleTable = ({
  customCell,
  customCellProps,
  loading = false,
}: {
  customCell?: React.ComponentType<any>;
  customCellProps?: object;
  loading?: boolean;
}): IRow[] => [
  {
    id: "front",
    isHeader: true,
    cells: [
      {
        id: "wawoo",
        value: "Wawooo!",
      },
      {
        id: "wah",
        value: "Wah!",
      },
      {
        id: "cool",
        value: "Cool!",
      },
    ],
  },
  {
    id: "front1",
    cells: [
      {
        id: "react",
        value: "React",
        cellContent: customCell,
        cellContentProps: customCellProps,
      },
      {
        id: "angular",
        value: "Angular",
      },
      {
        id: "vuejs",
        value: "VueJS",
      },
    ],
  },
  {
    id: "back",
    loading,
    cells: [
      {
        id: "python",
        value: "Python",
      },
      {
        id: "Go",
        value: "Go",
      },
      {
        id: "Java",
        value: "java",
      },
    ],
  },
  {
    id: "back1",
    cells: [
      {
        id: "python",
        value: "Python1",
      },
      {
        id: "Go",
        value: "Go",
      },
      {
        id: "Java",
        value: "java",
      },
    ],
  },
  {
    id: "back2",
    cells: [
      {
        id: "python",
        value: "Python2",
      },
      {
        id: "Go",
        value: "Go",
      },
      {
        id: "Java",
        value: "java",
      },
    ],
  },
  {
    id: "back3",
    cells: [
      {
        id: "python",
        value: "Python3",
      },
      {
        id: "Go",
        value: "Go",
      },
      {
        id: "Java",
        value: "java",
      },
    ],
  },
];

export const subRows = ({ subsubRows = [] }: { subsubRows?: IRow[] }): IRow[] => [
  {
    id: "foo",
    cells: [
      {
        id: "storybook",
        value: "Storybook",
      },
      {
        id: "jest",
        value: "Jest",
      },
      {
        id: "Enzym",
        value: "enzyme",
        subItems: subsubRows,
      },
    ],
  },
  {
    id: "back",
    cells: [
      {
        id: "docker",
        value: "Docker",
      },
      {
        id: "doccomp",
        value: "Docker compose",
      },
      {
        id: "Sentry",
        value: "Sentry",
        subItems: subsubRows,
      },
    ],
  },
];

export const subMiam: IRow[] = [
  {
    id: "miam",
    cells: [
      {
        id: "sushi",
        value: "Sushi",
      },
      {
        id: "pizza",
        value: "Pizza",
      },
      {
        id: "mafe",
        value: "Mafe",
      },
    ],
  },
];

export const tableWithSubItems = ({
  firstSubRows = [],
  secondSubRows = [],
}: {
  firstSubRows?: IRow[];
  secondSubRows?: IRow[];
}): IRow[] => {
  const table = [
    {
      id: "front",
      isHeader: true,
      isSelectable: false,
      cells: [
        {
          id: "wawoo",
          value: "Wawooo!",
        },
        {
          id: "wah",
          value: "Wah!",
        },
        {
          id: "cool",
          value: "Cool!",
        },
      ],
    },
    {
      title: "foo",
      id: "front1",
      cells: [
        {
          id: "react",
          value: "React",
          subItems: firstSubRows,
        },
        {
          id: "angular",
          value: "Angular",
        },
        {
          id: "vuejs",
          value: "VueJS",
        },
      ],
    },
    {
      id: "back",
      cells: [
        {
          id: "python",
          value: "Python",
        },
        {
          id: "Go",
          value: "Go",
          subItems: secondSubRows,
        },
        {
          id: "Java",
          value: "java",
        },
      ],
    },
  ];
  return table;
};

export const tableWithDifferentRowSizes: IRow[] = [
  {
    id: "front",
    isHeader: true,
    cells: [
      {
        id: "wawoo",
        value: "Wawooo!",
      },
      {
        id: "wah",
        value: "Wah!",
      },
      {
        id: "cool",
        value: "Cool!",
      },
    ],
  },
  {
    id: "front1",
    isHeader: false,
    cells: [
      {
        id: "foo",
        value: "foo",
      },
      {
        id: "bar",
        value: "bar",
      },
      {
        id: "fuzz",
        value: "fuzz",
      },
    ],
  },
  {
    id: "small",
    size: 25,
    cells: [
      {
        id: "small",
        value: "Wawooo!",
      },
      {
        id: "wah",
        value: "Wah!",
      },
      {
        id: "cool",
        value: "Cool!",
      },
    ],
  },
  {
    id: "medium",
    size: 40,
    cells: [
      {
        id: "medium",
        value: "React",
      },
      {
        id: "angular",
        value: "Angular",
      },
      {
        id: "vuejs",
        value: "VueJS",
      },
    ],
  },
  {
    id: "large",
    size: 80,
    cells: [
      {
        id: "large",
        value: "Python",
      },
      {
        id: "Go",
        value: "Go",
      },
      {
        id: "Java",
        value: "java",
      },
    ],
  },
];

export const tableWithDifferentColspan: IRow[] = [
  {
    id: "front",
    isHeader: true,
    cells: [
      {
        id: "wawoo",
        value: "Wawooo!",
      },
      {
        id: "wah",
        value: "Wah!",
      },
      {
        id: "cool",
        value: "Cool!",
      },
    ],
  },
  {
    id: "front1",
    isHeader: false,
    cells: [
      {
        id: "foo",
        value: "foo",
        colspan: 2,
      },
      {
        id: "fuzz",
        value: "fuzz",
      },
    ],
  },
  {
    id: "foo",
    cells: [
      {
        id: "wah",
        value: "Wah!",
      },
      {
        id: "cool",
        value: "Cool!",
        colspan: 2,
      },
    ],
  },
  {
    id: "foo",
    cells: [
      {
        id: "medium",
        value: "React",
      },
      {
        id: "angular",
        value: "Angular",
      },
      {
        id: "vuejs",
        value: "VueJS",
      },
    ],
  },
  {
    id: "foo",
    cells: [
      {
        id: "large",
        value: "Python",
        colspan: 3,
      },
    ],
  },
];

export const defaultColspanMatrix = (cellsCount: number) => [
  {
    cells: [
      {
        column: 0,
        value: "Table title",
        colspan: cellsCount,
        colspanMatrix: [
          {
            cells: [
              {
                column: 0,
                value: "****** My chart ******",
                colspan: cellsCount,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    cells: [
      {
        column: 1,
        colspan: 2,
      },
    ],
  },
];

export function generateTableWithCustomColspan(
  rowsCount: number,
  cellsCount: number,
  subRow = false,
  level = 0,
  colspanMatrix: any = defaultColspanMatrix(cellsCount)
) {
  const rows: IRow[] = Array.from(Array(rowsCount), (_, rowIndex) => {
    const totalColspan = colspanMatrix[rowIndex]
      ? colspanMatrix[rowIndex].cells.reduce((sum: number, currentValue: any) => {
          return sum + currentValue.colspan;
        }, 0)
      : 0;
    const cellsCountWithColspan =
      cellsCount - totalColspan + (colspanMatrix[rowIndex] ? colspanMatrix[rowIndex].cells.length : 0);
    const cells = generateRowWithCustomColumns(rowIndex, cellsCount, cellsCountWithColspan, subRow, level, colspanMatrix);

    return {
      id: `${rowIndex}`,
      isHeader: false,
      cells,
    };
  });

  return {
    id: `test`,
    rows,
  };
}

function generateRowWithCustomColumns(
  rowIndex: number,
  cellsCount: number,
  cellsCountWithColspan: number,
  subRow: boolean,
  level: number,
  colspanMatrix: any
) {
  return Array.from(Array(cellsCountWithColspan), (_, cellIndex) => {
    const currentCell = colspanMatrix[rowIndex]
      ? colspanMatrix[rowIndex].cells.find((cell: any) => cell.column === cellIndex && cell.colspan)
      : null;
    if (currentCell && currentCell.colspan >= 1) {
      return {
        id: `(${rowIndex},${cellIndex})-${level}`,
        value: currentCell.value || `(${rowIndex},${cellIndex})`,
        colspan: currentCell.colspan,
        subItems:
          subRow && cellIndex === 0
            ? Array.from(Array(cellsCount), (_, rowIndex) =>
                generateRow(rowIndex, cellsCount, level <= 2, level + 1, currentCell.colspanMatrix)
              )
            : [],
      };
    }
    return {
      id: `(${rowIndex},${cellIndex})-${level}`,
      value: `(${rowIndex},${cellIndex})`,
    };
  });
}

export function generateRow(index: number, cellsCount: number, subRow = false, level = 0, colspanMatrix?: any): IRow {
  const cells = Array.from(Array(cellsCount), (_, cellIndex) => {
    const currentCell =
      colspanMatrix && colspanMatrix[index]
        ? colspanMatrix[index].cells.find((cell: any) => cell.column === cellIndex && cell.colspan)
        : null;
    return {
      id: `(${index},${cellIndex})-${level}`,
      value: (currentCell && currentCell.value) || `(${index},${cellIndex})`,
      colspan: currentCell ? currentCell.colspan : 1,
      subItems:
        subRow && cellIndex === 0 && index === 2
          ? Array.from(Array(20), (_, rowIndex) => generateRow(rowIndex, cellsCount, level <= 2, level + 1))
          : [],
    };
  });
  return {
    id: `${index}`,
    isHeader: index === 0,
    cells,
  };
}

export function generateTable(rowsCount: number, cellsCount: number, props = {}, subRow = false) {
  const rows = Array.from(Array(rowsCount), (_, rowIndex) => generateRow(rowIndex, cellsCount, subRow));
  return {
    id: "table-foo",
    ...props,
    rows,
  };
}
