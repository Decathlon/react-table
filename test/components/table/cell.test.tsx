/// <reference path="../../typings/tests-entry.d.ts" />
import { createRenderer } from "react-test-renderer/shallow";
import { mount } from "enzyme";

import Cell from "../../../src/components/table/cell";
import { subRows } from "../../../stories/utils/tables";
import { withThemeProvider } from "../../../stories/utils/decorators";
import { MouseClickButtons } from "../../../src/components/constants";

const rows = subRows({});

describe("Cell component", () => {
  test("should render the default cell", () => {
    const props = { id: "foo", rowIndex: 0, relativeRowIndex: 0, value: "bar", subItems: [] };
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<Cell {...props} />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render the loading cell", () => {
    const props = { id: "foo", loading: true, rowIndex: 0, relativeRowIndex: 0, subItems: [] };
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<Cell {...props} />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render the cell with a custom class Name", () => {
    const props = {
      id: "foo",
      loading: true,
      rowIndex: 0,
      relativeRowIndex: 0,
      subItems: [],
      getClassName: () => "custom-className",
    };
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<Cell {...props} />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render a cell with custom style", () => {
    const props = {
      id: "foo",
      rowIndex: 0,
      relativeRowIndex: 0,
      value: "bar",
      subItems: [],
      style: {
        width: "50px",
        height: "25px",
      },
    };
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<Cell {...props} />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render a cell with custom style with colspan", () => {
    const props = {
      id: "foo",
      rowIndex: 0,
      relativeRowIndex: 0,
      value: "bar",
      subItems: [],
      colspan: 2,
      style: {
        width: 50,
        height: "25px",
      },
    };
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<Cell {...props} />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render a header cell", () => {
    const props = {
      id: "foo",
      rowIndex: 0,
      relativeRowIndex: 0,
      value: "bar",
      subItems: [],
    };
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<Cell {...props} component="th" />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render a cell with closed subItems", () => {
    const props = {
      id: "foo",
      rowIndex: 0,
      relativeRowIndex: 0,
      value: "bar",
      subItems: rows,
    };
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<Cell {...props} component="th" />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render a cell with subItems (hideSubItemsOpener)", () => {
    const props = {
      id: "foo",
      rowIndex: 0,
      relativeRowIndex: 0,
      value: "bar",
      subItems: rows,
      hideSubItemsOpener: true,
    };
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<Cell {...props} />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render a cell with opened subItems", () => {
    const props = {
      id: "foo",
      rowIndex: 0,
      relativeRowIndex: 0,
      value: "bar",
      subItems: rows,
      opened: true,
    };
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<Cell {...props} />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render a selected cell", () => {
    const props = {
      id: "foo",
      rowIndex: 0,
      relativeRowIndex: 0,
      value: "bar",
      isSelected: true,
    };
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<Cell {...props} />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render a cell with custom cell content", () => {
    const props = {
      id: "foo",
      rowIndex: 0,
      relativeRowIndex: 0,
      cellContent: ({ name }: { name: string }) => <div>{name}</div>,
      cellContentProps: { name: "Foo" },
    };
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<Cell {...props} />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should call mouse event callback", () => {
    const props = {
      id: "foo",
      index: 2,
      rowIndex: 0,
      relativeRowIndex: 0,
      value: "bar",
      subItems: rows,
      onMouseEnter: jest.fn(),
      onMouseDown: jest.fn(),
      onMouseUp: jest.fn(),
      onContextMenu: jest.fn(),
    };
    const wrapper = mount(withThemeProvider(() => <Cell {...props} />));
    // onMouseEnter
    wrapper.find("td").last().simulate("mouseenter");
    expect(props.onMouseEnter).toBeCalledWith({ rowIndex: 0, cellIndex: 2 });
    // onMouseDown
    wrapper
      .find("td")
      .last()
      .simulate("mousedown", { nativeEvent: { button: 0 } });
    expect(props.onMouseDown).toBeCalledWith({ rowIndex: 0, cellIndex: 2 }, MouseClickButtons.left);
    // onMouseUp
    wrapper.find("td").last().simulate("mouseup");
    expect(props.onMouseUp).toBeCalled();
    // onContextMenu
    const eventTarget = wrapper.find("[data-testid='table-cell-wrapper-foo']").last().getDOMNode();
    wrapper.find("td").last().simulate("contextmenu");
    expect(props.onContextMenu).toBeCalledWith({
      anchorEl: eventTarget,
      contextCell: { rowIndex: 0, cellIndex: 2 },
    });
  });

  test("should call onCallOpen callback", () => {
    const props = {
      id: "foo",
      index: 2,
      rowIndex: 0,
      relativeRowIndex: 0,
      value: "bar",
      subItems: rows,
      onCallOpen: jest.fn(),
    };
    const wrapper = mount(withThemeProvider(() => <Cell {...props} />));
    // onOpen
    wrapper.find("[data-testid='table-cell-sub-item-toggle']").last().simulate("click");
    expect(props.onCallOpen).toBeCalledWith(2);
  });
});
