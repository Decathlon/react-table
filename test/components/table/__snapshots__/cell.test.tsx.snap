// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell component should render a cell with closed subItems 1`] = `
<th
  className="table-column"
  colSpan={1}
  data-testid="table-column"
  onContextMenu={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "height": 56,
      "padding": 0,
      "width": "auto",
    }
  }
>
  <div
    className="table-overflow-wrapper"
    data-testid="table-cell-wrapper-foo"
    style={
      Object {
        "height": 56,
        "width": "auto",
      }
    }
  >
    <div
      className="table-cell-container"
      style={
        Object {
          "justifyContent": undefined,
        }
      }
    >
      <ForwardRef(WithStyles)
        className="table-cell-sub-item-toggle"
        data-testid="table-cell-sub-item-toggle"
        onClick={[Function]}
      >
        <ForwardRef(WithStyles)>
          keyboard_arrow_right
        </ForwardRef(WithStyles)>
      </ForwardRef(WithStyles)>
      <div
        className="cell-value"
        style={
          Object {
            "height": 56,
            "lineHeight": "56px",
          }
        }
        title="bar"
      >
        bar
      </div>
    </div>
  </div>
</th>
`;

exports[`Cell component should render a cell with custom cell content 1`] = `
<td
  className="table-column"
  colSpan={1}
  data-testid="table-column"
  onContextMenu={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "height": 56,
      "padding": 0,
      "width": "auto",
    }
  }
>
  <div
    className="table-overflow-wrapper"
    data-testid="table-cell-wrapper-foo"
    style={
      Object {
        "height": 56,
        "width": "auto",
      }
    }
  >
    <div
      className="table-cell-container"
      style={
        Object {
          "justifyContent": undefined,
        }
      }
    >
      <cellContent
        id="foo"
        index={0}
        name="Foo"
      />
    </div>
  </div>
</td>
`;

exports[`Cell component should render a cell with custom style 1`] = `
<td
  className="table-column"
  colSpan={1}
  data-testid="table-column"
  onContextMenu={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "height": "25px",
      "padding": 0,
      "width": "50px",
    }
  }
>
  <div
    className="table-overflow-wrapper"
    data-testid="table-cell-wrapper-foo"
    style={
      Object {
        "height": "25px",
        "width": "50px",
      }
    }
  >
    <div
      className="table-cell-container"
      style={
        Object {
          "justifyContent": undefined,
        }
      }
    >
      <div
        className="cell-value"
        style={
          Object {
            "height": "25px",
            "lineHeight": "25pxpx",
          }
        }
        title="bar"
      >
        bar
      </div>
    </div>
  </div>
</td>
`;

exports[`Cell component should render a cell with custom style with colspan 1`] = `
<td
  className="table-column"
  colSpan={2}
  data-testid="table-column"
  onContextMenu={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "height": "25px",
      "padding": 0,
      "width": 100,
    }
  }
>
  <div
    className="table-overflow-wrapper"
    data-testid="table-cell-wrapper-foo"
    style={
      Object {
        "height": "25px",
        "width": 100,
      }
    }
  >
    <div
      className="table-cell-container"
      style={
        Object {
          "justifyContent": undefined,
        }
      }
    >
      <div
        className="cell-value"
        style={
          Object {
            "height": "25px",
            "lineHeight": "25pxpx",
          }
        }
        title="bar"
      >
        bar
      </div>
    </div>
  </div>
</td>
`;

exports[`Cell component should render a cell with opened subItems 1`] = `
<td
  className="table-column"
  colSpan={1}
  data-testid="table-column"
  onContextMenu={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "height": 56,
      "padding": 0,
      "width": "auto",
    }
  }
>
  <div
    className="table-overflow-wrapper"
    data-testid="table-cell-wrapper-foo"
    style={
      Object {
        "height": 56,
        "width": "auto",
      }
    }
  >
    <div
      className="table-cell-container"
      style={
        Object {
          "justifyContent": undefined,
        }
      }
    >
      <ForwardRef(WithStyles)
        className="table-cell-sub-item-toggle"
        data-testid="table-cell-sub-item-toggle"
        onClick={[Function]}
      >
        <ForwardRef(WithStyles)>
          keyboard_arrow_down
        </ForwardRef(WithStyles)>
      </ForwardRef(WithStyles)>
      <div
        className="cell-value"
        style={
          Object {
            "height": 56,
            "lineHeight": "56px",
          }
        }
        title="bar"
      >
        bar
      </div>
    </div>
  </div>
</td>
`;

exports[`Cell component should render a cell with subItems (hideSubItemsOpener) 1`] = `
<td
  className="table-column"
  colSpan={1}
  data-testid="table-column"
  onContextMenu={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "height": 56,
      "padding": 0,
      "width": "auto",
    }
  }
>
  <div
    className="table-overflow-wrapper"
    data-testid="table-cell-wrapper-foo"
    style={
      Object {
        "height": 56,
        "width": "auto",
      }
    }
  >
    <div
      className="table-cell-container"
      style={
        Object {
          "justifyContent": undefined,
        }
      }
    >
      <div
        className="cell-value"
        style={
          Object {
            "height": 56,
            "lineHeight": "56px",
          }
        }
        title="bar"
      >
        bar
      </div>
    </div>
  </div>
</td>
`;

exports[`Cell component should render a header cell 1`] = `
<th
  className="table-column"
  colSpan={1}
  data-testid="table-column"
  onContextMenu={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "height": 56,
      "padding": 0,
      "width": "auto",
    }
  }
>
  <div
    className="table-overflow-wrapper"
    data-testid="table-cell-wrapper-foo"
    style={
      Object {
        "height": 56,
        "width": "auto",
      }
    }
  >
    <div
      className="table-cell-container"
      style={
        Object {
          "justifyContent": undefined,
        }
      }
    >
      <div
        className="cell-value"
        style={
          Object {
            "height": 56,
            "lineHeight": "56px",
          }
        }
        title="bar"
      >
        bar
      </div>
    </div>
  </div>
</th>
`;

exports[`Cell component should render a selected cell 1`] = `
<td
  className="table-column selected"
  colSpan={1}
  data-testid="table-column"
  onContextMenu={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "height": 56,
      "padding": 0,
      "width": "auto",
    }
  }
>
  <div
    className="table-overflow-wrapper"
    data-testid="table-cell-wrapper-foo"
    style={
      Object {
        "height": 56,
        "width": "auto",
      }
    }
  >
    <div
      className="table-cell-container"
      style={
        Object {
          "justifyContent": undefined,
        }
      }
    >
      <div
        className="cell-value"
        style={
          Object {
            "height": 56,
            "lineHeight": "56px",
          }
        }
        title="bar"
      >
        bar
      </div>
    </div>
  </div>
</td>
`;

exports[`Cell component should render the default cell 1`] = `
<td
  className="table-column"
  colSpan={1}
  data-testid="table-column"
  onContextMenu={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "height": 56,
      "padding": 0,
      "width": "auto",
    }
  }
>
  <div
    className="table-overflow-wrapper"
    data-testid="table-cell-wrapper-foo"
    style={
      Object {
        "height": 56,
        "width": "auto",
      }
    }
  >
    <div
      className="table-cell-container"
      style={
        Object {
          "justifyContent": undefined,
        }
      }
    >
      <div
        className="cell-value"
        style={
          Object {
            "height": 56,
            "lineHeight": "56px",
          }
        }
        title="bar"
      >
        bar
      </div>
    </div>
  </div>
</td>
`;

exports[`Cell component should render the loading cell 1`] = `
<td
  className="table-column"
  colSpan={1}
  data-testid="table-column"
  onContextMenu={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "height": 56,
      "padding": 0,
      "width": "auto",
    }
  }
>
  <div
    className="table-overflow-wrapper"
    data-testid="table-cell-wrapper-foo"
    style={
      Object {
        "height": 56,
        "width": "auto",
      }
    }
  >
    <div
      className="table-cell-container"
      style={
        Object {
          "justifyContent": undefined,
        }
      }
    >
      <div
        className="cell-value"
        style={
          Object {
            "height": 56,
            "lineHeight": "56px",
          }
        }
        title=""
      >
        <div
          className="cell-skeleton-container"
          style={
            Object {
              "height": 56,
              "lineHeight": "56px",
            }
          }
        >
          <ForwardRef(WithStyles)
            height={15}
            variant="rect"
            width={30}
          />
        </div>
      </div>
    </div>
  </div>
</td>
`;
